DVOP-2124 v16 centos:8 to amazonlinux:2

autmwatl opened the pull request:
centos8 has been deprecated. migrate to amazonlinux2

Commit:
DVOP-2124 centos:8 to amazonlinux:2
centos8 has been deprecated. migrate to amazonlinux2

Commit:
DVOP-2124 amend yum command
yum module not compatible with amazon linux. amend to install node directly

sienkin commented on Dockerfile:
```diff
@@ -1,12 +1,8 @@
 RUN yum update -y && \
-    yum module install nodejs:16/development -y && \
+    curl --silent --location https://rpm.nodesource.com/setup_16.x  | bash - && \
+    yum -y install nodejs && \
```
Line 5 is superfluous as `nodejs` is already included in the list of packages to be installed.

Commit:
DVOP-2124 remove duplicated install
remove installation of node as it is being installed in the subsequent yum install command

sienkin commented on Dockerfile:
```diff
@@ -1,12 +1,7 @@
+FROM amazonlinux:2
 
 RUN yum update -y && \
-    yum module install nodejs:16/development -y && \
+    curl --silent --location https://rpm.nodesource.com/setup_16.x  | bash - && \
```
The setup script installs a local repository configuration. As a result, it would be beneficial to have this executed before the `yum update -y && \` line. Best to have it on its own line as it's not directly related to the `yum` command. Similar to the way the v10 dockerfile is configured.

E.g.:
```
RUN curl --silent --location https://rpm.nodesource.com/setup_16.x | bash -

RUN yum update -y && \
```

Also, the `nss` package is installed by default in Amazon Linux 2 so can be omitted from the lsit of packages.

Commit:
DVOP-2124 Remove nss, separate commands
The nss package is installed by default in Amazon Linux 2. Omit from the list of packages to be installed.
Separate curl and update into separate execution lines for optimisation.

sienkin commented on Dockerfile:
```diff
@@ -1,12 +1,7 @@
-
+FROM amazonlinux:2
 
+RUN curl --silent --location https://rpm.nodesource.com/setup_16.x  | bash - && \
```
You don't need the trailing `&& \` here as this command is on its own and doesn't lead in to any others

Commit:
DVOP-2124 Remove unnecessary trailing &&
This is no additional command. Remove trailing &&
